msdbook/tests/test_fishery_dynamics.py

```diff
@@ -0,0 +1,81 @@
+import pytest
+import numpy as np 
+import matplotlib.pyplot as plt
+from mpl_toolkits.mplot3d import Axes3D  # Import Axes3D
+from matplotlib.testing.decorators import check_figures_equal
+from msdbook.fishery_dynamics import plot_objective_performance, plot_factor_performance
+
+@pytest.fixture
+def sample_data():
+    """Fixture to provide sample data for testing."""
+    objective_performance = np.array([
+        [1, 2, 3, 4, 5],
+        [2, 3, 4, 5, 6],
+        [3, 4, 5, 6, 7]
+    ])
+    profit_solution = 1
+    robust_solution = 2
+    param_values = np.random.rand(100, 7)  # Example data
+    collapse_days = np.random.rand(100, 2) * 10  # Example data
+    b = np.linspace(0, 1, 10)
+    m = np.linspace(0, 1, 10)
+    a = np.linspace(0, 2, 10)
+    return {
+        'objective_performance': objective_performance,
+        'profit_solution': profit_solution,
+        'robust_solution': robust_solution,
+        'param_values': param_values,
+        'collapse_days': collapse_days,
+        'b': b,
+        'm': m,
+        'a': a
+    }
+
+@pytest.mark.filterwarnings("ignore::matplotlib.MatplotlibDeprecationWarning")
```

Comment:
Is this throwing a depreciation error without the filter?

Reply:
No it just gives 2 warnings for both test_plot_objective_performance and test_plot_objective_performance without the filter.

Here is the warnings it showed without the filter:
msdbook/tests/test_fishery_dynamics.py::test_plot_objective_performance
  /Users/lari627/msd_uncertainty_ebook/msdbook/fishery_dynamics.py:51: MatplotlibDeprecationWarning: The get_cmap function was deprecated in Matplotlib 3.7 and will be removed in 3.11. Use ``matplotlib.colormaps[name]`` or ``matplotlib.colormaps.get_cmap()`` or ``pyplot.get_cmap()`` instead.
    cmap = plt.cm.get_cmap("Blues")

msdbook/tests/test_fishery_dynamics.py::test_plot_factor_performance
  /Users/lari627/msd_uncertainty_ebook/msdbook/fishery_dynamics.py:134: MatplotlibDeprecationWarning: The get_cmap function was deprecated in Matplotlib 3.7 and will be removed in 3.11. Use ``matplotlib.colormaps[name]`` or ``matplotlib.colormaps.get_cmap()`` or ``pyplot.get_cmap()`` instead.
    cmap = plt.cm.get_cmap("RdBu_r")

Reply:
Gotcha, you should address those depreciation issues.

Reply:
I updated the code it removed the comments and added the pytest.warns context manager now captures the DeprecationWarning while running the plotting functions, which helps to suppress the warnings during the tests.

Reply:
ah ok, I missed that the function you're testing is throwing the depreciation errors. Can you roll back the depreciation error catch, it's good to see those errors. I've created a new [issue](https://github.com/IMMM-SFA/msd_uncertainty_ebook/issues/122) to address the depreciation errors.